api/server/router/network/network_routes.go

```diff
@@ -275,12 +275,7 @@ func (n *networkRouter) postNetworkConnect(ctx context.Context, w http.ResponseW
 		return err
 	}
 
-	// Always make sure there is no ambiguity with respect to the network ID/name
```

Comment:
Was this part of the problem? Sorry, having trouble understanding this change 😅 (this was added in https://github.com/moby/moby/commit/cafed80cd019a8b40025eaa5e5b37459362607fb)

Reply:
Thanks @thaJeztah for the review. 

I added comments in the code to explain the reason for non-ambiguity check. Basically at the time of `connect` the attachable network may not be available so resolving name to ID will not work. On the other hand, the name collision check will be handled by the daemon in this case anyway so the check has been removed.

The PR has been updated.